pkg/stanza/fileconsumer/internal/reader/factory.go

```diff
@@ -119,5 +122,17 @@ func (f *Factory) NewReaderFromMetadata(file *os.File, m *Metadata) (r *Reader,
 	for k, v := range attributes {
 		r.FileAttributes[k] = v
 	}
+
+	if strings.HasSuffix(file.Name(), ".gz") {
```

Comment:
I'd like to isolate any new functionality, at least for now. This is both to ensure expected behavior is not affected but also performance is not degraded for use cases we already support. Can we add a config option instead which calls out the compression format that should be assumed? Potentially later we could add an "auto" mode which evaluates the format and reacts accordingly.